inspect.go

```diff
@@ -0,0 +1,166 @@
+package enmime
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"mime"
+	"net/textproto"
+	"strings"
+
+	"github.com/jhillyerd/enmime/internal/coding"
+	"github.com/pkg/errors"
+)
+
+func HumanHeadersOnly(b []byte) (map[string]string, error) {
+	b = clean(b)
+	tr := textproto.NewReader(bufio.NewReader(bytes.NewReader(b)))
+	headers, err := tr.ReadMIMEHeader()
+	switch errors.Cause(err) {
+	case nil, io.EOF:
+	// carry on, io.EOF is expected
+	default:
+		return nil, err
+	}
+	bs := bufio.NewScanner(bufio.NewReader(bytes.NewReader(b)))
```

Comment:
Scanner has it's own buffer, don't need the extra bufio.Reader.

Reply:
Not preserving order for this use case, so scanner isn't required anymore